src/query.rs

```diff
@@ -404,38 +403,43 @@ impl Query {
     // HD Wallets
     //
 
-    pub fn get_hd_wallets(&self) -> HashMap<Fingerprint, HDWallet> {
+    pub fn get_hd_wallets(&self) -> HashMap<DescriptorChecksum, HDWallet> {
         self.indexer.read().unwrap().watcher().wallets().clone()
     }
 
-    pub fn get_hd_wallet(&self, fingerprint: Fingerprint) -> Option<HDWallet> {
+    pub fn get_hd_wallet(&self, desc_check: DescriptorChecksum) -> Option<HDWallet> {
         self.indexer
             .read()
             .unwrap()
             .watcher()
-            .get(fingerprint)
+            .get(desc_check)
             .cloned()
     }
 
     // get the ScriptInfo entry of a derived hd key, without it necessarily being indexed
-    pub fn get_hd_script_info(&self, fingerprint: Fingerprint, index: u32) -> Option<ScriptInfo> {
+    pub fn get_hd_script_info(
+        &self,
+        desc_check: DescriptorChecksum,
+        index: u32,
+    ) -> Option<ScriptInfo> {
         let indexer = self.indexer.read().unwrap();
-        let wallet = indexer.watcher().get(fingerprint)?;
-        let key = wallet.derive(index);
-        let address = wallet.to_address(&key);
+        let wallet = indexer.watcher().get(desc_check.clone());
+        let address = wallet.unwrap().derive_address(index, &self.rpc).unwrap();
```

Comment:
The first `unwrap()` should instead result in returning an `Ok(None)` (can be done with the [`some_or_ret!` macro](https://github.com/shesek/bwt/blob/af4e3970400734c13e74f81039b3872c5319a609/src/macros.rs#L80), search for it in the codebase to find example usage), the second `unwrap()` should instead result in returning an `Err` (can be done with the `?` operator).